### ISSUE TITLE
redirectTo does not seem to work (because it's forbidden)

### ISSUE THREAD

**Issue by laurentS** [supabase/gotrue-js#65]
# Bug report

## Describe the bug

I've been going a little insane over this `redirectTo` parameter with auth calls. Whatever I tried passing was not getting sent to the server, despite the fix for #63.
It turns out that setting the `Referer` header programmatically is not allowed: https://developer.mozilla.org/en-US/docs/Glossary/Forbidden_header_name

## To Reproduce

See the steps in #63, but upgrade supabase-js to 1.4.1.
The magic link `redirect_to` parameter contains the calling url, not the one passed to gotrue's `redirectTo`.
A warning is issued in firefox's console: `Attempt to set a forbidden header was denied: referer`

## Expected behavior

The redirection link in the email should match what is passed to the auth call. I'd suggest either passing a different, custom header, or passing the url as part of the body of the request, whichever requires the least changes on the server side.

## System information

- OS: debian/linux
- Browser firefox 84
- Version of supabase-js: 1.4.1
- gotrue-js version 1.11.2

**Comment by kiwicopple** [comment]
Hey @laurentS, can I please confirm that you are trying to redirect to a path that is the same as the URL/origin in your auth Settings?

From the [docs](https://supabase.io/docs/client/auth-signup#sign-up-with-redirect):

> the base domain must be the same as the domain in your Auth settings

So for settings like this:

![image](https://user-images.githubusercontent.com/10214025/108954270-851fc580-76a7-11eb-86ce-e8c19987cb4b.png)


I can redirect to:

- `https://supabase.io/welcome`
- `https://supabase.io/magic`

But I cannot redirect to:

- `https://evil.io/welcome`


We will eventually enable multiple origins: https://github.com/supabase/gotrue/issues/63

**Comment by laurentS** [comment]
@kiwicopple yes, correct. I have `http://localhost:3000/` set as the site URL in the auth settings (I had a different one initially, like `http://localhost:3000/sendmehere` but all on the same domain), and I'm trying to redirect to `http://localhost:3000/someotherurl`. Having logged the various calls through the supabase stack, I can see that the redirect url I'm setting "someotherurl" passed all the way down to `params` here https://github.com/supabase/gotrue-js/blob/d993b5457e8c54d4e9b6d4ff221e28c1adc2964d/src/lib/fetch.ts#L37

This what `params` looks like:
```json
{
  "method": "POST",
  "headers": {
    "Content-Type": "text/plain;charset=UTF-8",
    "Authorization": "Bearer exxxxxxo",
    "apikey": "myapikey",
    "referer": "http://localhost:3000/someotherurl"
  },
  "body": "{\"email\":\"me@example.com\"}"
}
```
And the firefox dev tools show a different url in `Referer`, along with the warning mentioned above about trying to set a forbidden header:
![image](https://user-images.githubusercontent.com/1745184/108987677-77f4dd80-768b-11eb-8161-f69985421cc5.png)

So my understanding is that this is not related to what is set in the auth settings, but rather that you're trying to do something the browser won't let you do.

**Comment by kiwicopple** [comment]
ugh, yeah the `referer` header is messy, but unfortunately it is what's in the GoTrue server. I think we will need to seriously overhaul our fork if we want to move away from this method.

Let me discuss with @awalias and get back to you

**Comment by awalias** [comment]
I suppose we could additionally set redirect_to everywhere and if it's present choose it over referer header on the backend

**Comment by awalias** [comment]
we can now start setting redirect_to as a query param and it will be prioritized over the redirect header by the backend
